File: isitquesoweek.go
package main

import (
    "fmt"
    "time"
    "flag"
)

func IsQuesoWeek(day time.Time) (bool) {
    dow := int(day.Weekday())
    bow := day.AddDate(0, 0, -(dow))
    for bow.Month() == day.Month() {
        bow = bow.AddDate(0,0,-7)
    }

    queso_week_start := bow.AddDate(0,0,14)
    queso_week_end := queso_week_start.AddDate(0,0,8)

    if day.After(queso_week_start) && day.Before(queso_week_end) {
        return true
    } else {
        return false
    }
}

func main() {
    const layout = "Jan 02, 2006"
    all := flag.Bool("a", false, "shows all days of the year and which fall in queso week")
    flag.Parse()

    if *all {
        this_year := time.Now().Year()
        for d := time.Date(this_year,1,1,0,0,0,0,time.Local); d.Before(time.Date(this_year+1,1,1,0,0,0,0,time.Local)); d = d.AddDate(0,0,1) {
            fmt.Printf("%s: %t\n", d.Format(layout), IsQuesoWeek(d))
        }
    } else {
        if IsQuesoWeek(time.Now()) {
            fmt.Printf("Yep!")
        } else {
            fmt.Printf("Nope")
        }
    }
}

File: isitquesoweek.py
#!/usr/bin/python

import sys
from datetime import datetime, timedelta
import argparse

def is_queso_week(today=None):
    if today is None:
        today = datetime.now()
    #get beginning of the week (monday)
    bow = today - timedelta(today.weekday())
    #go back week by week until we get to the previous month's monday
    while(bow.month == today.month):
        bow = bow - timedelta(7)
    #go forward two weeks to get the second monday of the month
    valid_second_week = bow + timedelta(14)
    #does the requested date fit in this week?
    if today >= valid_second_week and today < (valid_second_week + timedelta(7)):
        return True
    else:
        return False

if __name__ == '__main__':
    parser = argparse.ArgumentParser()
    parser.add_argument('-a', action='store_true', help="show all days in the year that fall in queso week")
    args = parser.parse_args()
    today = datetime.now()
    if args.a:
        x = datetime(datetime.now().year, 1, 1)
        while(x.year == today.year):
            print "%s: %s" % (x.strftime("%b %d, %Y"), is_queso_week(x))
            x += timedelta(days=1)
    else:
        if is_queso_week(today):
            print "Yep!"
        else:
            print "Nope"
